### PR TITLE
Normalize options.method

### PR DESCRIPTION
**Pull request by szmarczak**

### CODE REVIEW

File: source/index.js [hunk 1]
```diff
@@ -33,7 +33,6 @@ const defaults = {
 		},
 		cache: false,
 		decompress: true,
-		useElectronNet: false,
```
Comment (sindresorhus) [inline source/index.js]: Why is this removed?

Comment (szmarczak) [reply source/index.js]: Because it's not needed. See https://github.com/sindresorhus/got/blob/master/source/request-as-event-emitter.js#L43

Comment (sindresorhus) [reply source/index.js]: I think it's still useful to declare all options up-front, even if they are `false` by default.

File: source/create.js [hunk 2]
```diff
@@ -37,7 +37,8 @@ const create = defaults => {
 		return defaults.handler(normalizeArguments(url, options, defaults), next);
 	};
 
-	for (const method of defaults.methods) {
+	const methods = defaults.methods.map(method => method.toLowerCase());
```
Comment (sindresorhus) [inline source/create.js]: Why are you lowercasing them here, but uppercasing them in `normalize-arguments`?

File: source/index.js (hunk 1 above)
Comment (szmarczak) [reply source/index.js]: Ok, I'll leave that then.

File: source/create.js (hunk 2 above)
Comment (szmarczak) [reply source/create.js]: > Why are you lowercasing them here,

Imagine calling `got.POST` instead of `got.post` :P

> but uppercasing them in normalize-arguments?

To keep things consistent; because `error.method` is uppercased :)

Comment (sindresorhus) [reply source/create.js]: Nvm. I missed the context. It's late here...

Comment (szmarczak) [reply source/create.js]: That's ok, answering questions is my job :P It's a bit late here too :)

File: source/index.js (hunk 1 above)
Comment (szmarczak) [reply source/index.js]: Done.
